src/tcgwars/logic/impl/gen3/LegendMaker.groovy

```diff
@@ -2248,15 +2248,23 @@ public enum LegendMaker implements LogicCardInfo {
           }
           eff2 = delayed {
             boolean flag = false
+            def flaggedPokemon
```

Comment:
You should be able to use the single flag for both EVOLVE and DEVOLVE effects since they won't happen at the same time.

Reply:
Not the same kind. for evolve you can use `ef.pokemonToBeEvolved`. In devolved a workaround is used, with a variable pointing to the PCSs and checking those not the same. It's too many checks for this one (and flaggedPokemon should be removed once a way to point to "pokemonToBeDevolved" is known/added).

Reply:
(PS: Didn't notice I commited this under this PR, was meant to be a direct commit.)

Reply:
Oh, my bad, didn't realize it was holding a PCS in it.